lib/mpool/src/mdc.c

```diff
@@ -29,6 +29,9 @@ struct mpool_mdc {
     struct mdc_file *mfpa;
 };
 
+/* Forward decls */
+merr_t
+mpool_mdc_delete(struct mpool *mp, uint64_t logid1, uint64_t logid2);
```

Comment:
This file provides `mpool_mdc_delete` for external use, so it must exist in a header file.  Why not include that header file here?  It's good practice to do that to ensure function prototypes match.   If mpool.h has too much crud and you would rather not include it here, then move the mpool_mdc family of functions to their own header file.

Reply:
I think the mpool "public interface" is probably due for a little reorganization like a lot of HSE in my opinion. Candidate for more build system improvements that I am itching to test out with the "hse config enhancements" ticket which is assigned to me.

Reply:
I had a mocking related compilation failure when including <mpool/mpool.h>. Yet another place that I wanted to revisit after opening this PR, will fix.

Reply:
Go to bed! :sleeping:

Reply:
Almost done :-) Will deal with mongo tomorrow....